deepface/modules/recognition.py

```diff
@@ -194,8 +192,8 @@ def find(
         )
 
     # append replaced images into both old and new images. these will be dropped and re-added.
-    new_images = new_images + replaced_images
-    old_images = old_images + replaced_images
+    new_images.update(replaced_images)
```

Comment:
suppose that `refresh_database  = False`, new_images and old_images are not initialized anymore in line 163 and 164. Line 195 and 196 will throw expception in that case.

Additionally, we should have an unit test for refresh_database = False